server/services/notify/notifymentions/mentions_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package notifymentions
+
+import (
+	"reflect"
+	"testing"
+
+	mm_model "github.com/mattermost/mattermost-server/v6/model"
+)
+
+var (
+	user1 = &mm_model.User{
+		Id:       mm_model.NewId(),
+		Username: "dlauder",
+	}
+	user2 = &mm_model.User{
+		Id:       mm_model.NewId(),
+		Username: "steve.mqueen",
+	}
+	user3 = &mm_model.User{
+		Id:       mm_model.NewId(),
+		Username: "bart_",
+	}
+
+	mockUsers = map[string]*mm_model.User{
+		"dlauder":      user1,
+		"steve.mqueen": user2,
+		"bart_":        user3,
+	}
+)
+
+func Test_userFromUsername(t *testing.T) {
+	delivery := newDeliveryMock(mockUsers)
+
+	tests := []struct {
+		name    string
+		uname   string
+		want    *mm_model.User
+		wantErr bool
+	}{
+		{name: "user1", uname: user1.Username, want: user1, wantErr: false},
+		{name: "user1 with period", uname: user1.Username + ".", want: user1, wantErr: false},
+		{name: "user1 with period plus more", uname: user1.Username + ". ", want: user1, wantErr: false},
+		{name: "user2 with periods", uname: user2.Username + "...", want: user2, wantErr: false},
+		{name: "user2 with underscore", uname: user2.Username + "_", want: user2, wantErr: false},
+		{name: "user2 with hyphen plus more", uname: user2.Username + "- ", want: user2, wantErr: false},
+		{name: "user2 with hyphen plus all", uname: user2.Username + ".-_ ", want: user2, wantErr: false},
+		{name: "user3 with underscore", uname: user3.Username + "_", want: user3, wantErr: false},
```

Comment:
I don't see any error tests. Should a test be added for user4? (wantErr: true)

Reply:
Done via https://github.com/mattermost/focalboard/pull/1147/commits/1c131c9f301a8e2cc9e5258516de9473d0533e73